pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -120,11 +119,7 @@ func (c *Controller) Services() ([]*model.Service, error) {
 		// Race condition: multiple threads may call Services, and multiple services
 		// may modify one of the service's cluster ID
 		clusterAddressesMutex.Lock()
-		if r.Cluster() == "" { // Should we instead check for registry name to be on safe side?
-			// If the service does not have a cluster ID (ServiceEntries, CloudFoundry, etc.)
-			// Do not bother checking for the cluster ID.
-			// DO NOT ASSIGN CLUSTER ID to non-k8s registries. This will prevent service entries with multiple
-			// VIPs or CIDR ranges in the address field
+		if r.Provider() == serviceregistry.External {
```

Comment:
should this be `r.Provider() != serviceregistry.Kubernetes` to match the logic in the other function?

Reply:
Here in particular I'm not sure if we want to check that... things that fall under this conditional will _not_ have clusterVips merged. Not sure if that's intended for all non-kube registries or only serviceeentry.